p2p/discovery/mdns.go

```diff
@@ -109,37 +109,42 @@ func (m *mdnsService) Close() error {
 }
 
 func (m *mdnsService) pollForEntries() {
+	entriesCh := make(chan *mdns.ServiceEntry, 16)
+	go func() {
+		for entry := range entriesCh {
+			m.handleEntry(entry)
+		}
+	}()
+
 	ticker := time.NewTicker(m.interval)
 	for range ticker.C {
-		entriesCh := make(chan *mdns.ServiceEntry, 16)
-		go func() {
-			for entry := range entriesCh {
-				m.handleEntry(entry)
-			}
-		}()
```

Comment:
this is incorrect-- how does it pass tests? 

this chan creation and goroutine is here because there may be `mdns.Query` errors for a single peer. the way you moved it out to the top of `pollForEntries` will make it so that it will close the chan exit after the first error-- and it may even call `close(entriesCh)` multiple times (!!!). 

move the chan creation + goroutine back to within the for loop.